bindings/go/reader.go

```diff
@@ -0,0 +1,299 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package opendal
+
+import (
+	"context"
+	"io"
+	"unsafe"
+
+	"github.com/jupiterrider/ffi"
+	"golang.org/x/sys/unix"
+)
+
+// Read reads the entire contents of the file at the specified path into a byte slice.
+//
+// This function is a wrapper around the C-binding function `opendal_operator_read`.
+//
+// # Parameters
+//
+//   - path: The path of the file to read.
+//
+// # Returns
+//
+//   - []byte: The contents of the file as a byte slice.
+//   - error: An error if the read operation fails, or nil if successful.
+//
+// # Notes
+//
+//   - This implementation does not support the `read_with` functionality.
+//   - Read allocates a new byte slice internally. For more precise memory control
+//     or lazy reading, consider using the Reader() method instead.
+//
+// # Example
+//
+//	func exampleRead(op *opendal.Operator) {
+//		data, err := op.Read("test")
+//		if err != nil {
+//			log.Fatal(err)
+//		}
+//		fmt.Printf("Read: %s\n", data)
+//	}
+//
+// Note: This example assumes proper error handling and import statements.
+func (op *Operator) Read(path string) ([]byte, error) {
+	read := getFFI[operatorRead](op.ctx, symOperatorRead)
+	bytes, err := read(op.inner, path)
+	if err != nil {
+		return nil, err
+	}
+
+	data := parseBytes(bytes)
+	if len(data) > 0 {
+		free := getFFI[bytesFree](op.ctx, symBytesFree)
+		free(bytes)
+
+	}
+	return data, nil
+}
+
+// Reader creates a new Reader for reading the contents of a file at the specified path.
+//
+// This function is a wrapper around the C-binding function `opendal_operator_reader`.
+//
+// # Parameters
+//
+//   - path: The path of the file to read.
+//
+// # Returns
+//
+//   - *OperatorReader: A reader for accessing the file's contents. It implements `io.ReadCloser`.
+//   - error: An error if the reader creation fails, or nil if successful.
+//
+// # Notes
+//
+//   - This implementation does not support the `reader_with` functionality.
+//   - The returned reader allows for more controlled and efficient reading of large files.
+//
+// # Example
+//
+//	func exampleReader(op *opendal.Operator) {
+//		r, err := op.Reader("path/to/file")
+//		if err != nil {
+//			log.Fatal(err)
+//		}
+//		defer r.Close()
+//
+//		size := 1024 // Read 1KB at a time
+//		buffer := make([]byte, size)
+//
+//		for {
+//			n, err := r.Read(buffer)
+//			if err != nil {
+//				log.Fatal(err)
+//			}
+//			fmt.Printf("Read %d bytes: %s\n", n, buffer[:n])
+//		}
+//	}
+//
+// Note: This example assumes proper error handling and import statements.
+func (op *Operator) Reader(path string) (*OperatorReader, error) {
+	getReader := getFFI[operatorReader](op.ctx, symOperatorReader)
+	inner, err := getReader(op.inner, path)
+	if err != nil {
+		return nil, err
+	}
+	reader := &OperatorReader{
+		inner: inner,
+		op:    op,
+	}
+	return reader, nil
+}
+
+type OperatorReader struct {
```

Comment:
Can we just use `Reader`? User will use this as `opendal.Reader`, right?

Reply:
Make sense.